centreon/src/Core/Security/Authentication/Infrastructure/Provider/OpenId.php

```diff
@@ -313,4 +316,55 @@ public function getAclConditionsMatches(): array
     {
         return $this->provider->getAclConditionsMatches();
     }
+
+    public function getTokenForSession(): ?string
+    {
+        return $this->provider->getTokenForSession();
+    }
+
+    /**
+     * Redirect the user to the OIDC end-session endpoint
+     *
+     * @param string $idToken
+     *
+     * @throws Exception
+     */
+    public function logout(string $idToken): void
+    {
+        $request = $this->requestStack->getCurrentRequest();
+        if ($request === null) {
+            throw new Exception('Request is not available for OpenID logout');
+        }
+
+        /** @var CustomConfiguration $customConfig */
+        $customConfig = $this->provider->getConfiguration()->getCustomConfiguration();
+        $baseUrl = $customConfig->getBaseUrl();
+        $endSessionEndpoint = $customConfig->getEndSessionEndpoint();
+
+        if (empty($baseUrl) || empty($endSessionEndpoint)) {
+            throw new Exception('Missing required OpenID configuration for logout');
+        }
+
+        $endSessionUrl = $baseUrl . $endSessionEndpoint;
+        $centreonBase = $request->getSchemeAndHttpHost() . $request->getBaseUrl();
+        $postLogout = $centreonBase . '/login';
+
+        $params = [
+            'post_logout_redirect_uri' => $postLogout,
+            'id_token_hint' => $idToken,
+        ];
+
+        $logoutUrl = $endSessionUrl . '?' . http_build_query($params);
+
+        try {
+            header('Location: ' . $logoutUrl, true, 302);
+            if (function_exists('fastcgi_finish_request')) {
+                fastcgi_finish_request();
+            }
+
+            exit;
+        } catch (Exception $e) {
+            throw new Exception('Failed to redirect to logout URL: ' . $e->getMessage());
```

Comment:
the previous is missing
```suggestion
            throw new Exception('Failed to redirect to logout URL: ' . $e->getMessage(), previous: $ex);
```